### PR TITLE
refactor: migrate lacework_integration_ecr to use v2 api

### PR DESCRIPTION
**Pull request by dmurray-lacework**
Signed-off-by: Darren Murray <darren.murray@lacework.net>

***Issue***:  https://lacework.atlassian.net/browse/ALLY-1138

***Description:***
Migrate CRUD of lacework_integration_ecr to use v2 api

***Additional Info:***
relies on go-sdk changes -> https://github.com/lacework/go-sdk/pull/904

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -364,8 +364,9 @@ github.com/zclconf/go-cty v1.1.0/go.mod h1:xnAOWiHeOqg2nWS62VtQ7pbOu17FtxJNW8RLE
 github.com/zclconf/go-cty v1.2.0/go.mod h1:hOPWgoHbaTUnI5k4D2ld+GRpFJSCe6bCM7m1q/N4PQ8=
 github.com/zclconf/go-cty v1.8.0/go.mod h1:vVKLxnk3puL4qRAv72AO+W99LUD4da90g3uUAzyuvAk=
 github.com/zclconf/go-cty v1.8.1/go.mod h1:vVKLxnk3puL4qRAv72AO+W99LUD4da90g3uUAzyuvAk=
-github.com/zclconf/go-cty v1.10.0 h1:mp9ZXQeIcN8kAwuqorjH+Q+njbJKjLrvB2yIh4q7U+0=
 github.com/zclconf/go-cty v1.10.0/go.mod h1:vVKLxnk3puL4qRAv72AO+W99LUD4da90g3uUAzyuvAk=
+github.com/zclconf/go-cty v1.11.0 h1:726SxLdi2SDnjY+BStqB9J1hNp4+2WlzyXLuimibIe0=
+github.com/zclconf/go-cty v1.11.0/go.mod h1:s9IfD1LK5ccNMSWCVFCE2rJfHiZgi7JijgeWIMfhLvA=
```
Comment (rmoles) [inline go.sum:369]: Should we have 2 versions of this. I see v1.10.0 is on line 367

File: integration/integration.go [hunk 2]
```diff
@@ -118,15 +118,15 @@ func GetIntegrationName(result string, integration string) string {
 	return res.Data.Name
 }
 
-func GetEcrWithCrossAccountCreds(result string) api.AwsEcrWithCrossAccountIntegration {
+func GetEcrWithCrossAccountCreds(result string) api.AwsEcrIamRoleIntegration {
```
Comment (rmoles) [inline integration/integration.go:121]: curious as to the change from `api.AwsEcrWithCrossAccountIntegration` to `api.AwsEcrIamRoleIntegration`.

File: go.sum (hunk 1 above)
Comment (dmurray-lacework) [reply go.sum:369]: We're not using 2 versions, this is the go.sum file.
https://go.dev/ref/mod#go-sum-files

File: integration/integration.go (hunk 2 above)
Comment (dmurray-lacework) [reply integration/integration.go:121]: api.AwsEcrWithCrossAccountIntegration is the v1 implementation

```
type AwsEcrWithCrossAccountIntegration struct {
	commonIntegrationData
	Data AwsEcrDataWithCrossAccountCreds `json:"DATA"`
}

type AwsEcrDataWithCrossAccountCreds struct {
	Credentials AwsCrossAccountCreds `json:"CROSS_ACCOUNT_CREDENTIALS" mapstructure:"CROSS_ACCOUNT_CREDENTIALS"`
	AwsEcrCommonData
}
```

File: lacework/resource_lacework_integration_ecr.go [hunk 3]
```diff
@@ -544,50 +367,40 @@ func resourceLaceworkIntegrationEcrUpdateWithAccessKey(d *schema.ResourceData, l
...
-
-	log.Printf("[INFO] Verifying server response")
-	err = validateEcrWithAccessKeyIntegrationResponse(&response)
+	log.Printf("[INFO] Updating %s registry\n", api.AwsEcrContainerRegistry.String())
+	response, err := lacework.V2.ContainerRegistries.UpdateAwsEcrAccessKey(data)
 	if err != nil {
 		return err
 	}
 
 	// @afiune at this point in time, we know the data field has a single value
```
Comment (rmoles) [inline lacework/resource_lacework_integration_ecr.go:376]: Do we still need this comment?

Comment (dmurray-lacework) [reply lacework/resource_lacework_integration_ecr.go:376]: 👍 I'll remove it

File: integration/integration.go (hunk 2 above)
Comment (rmoles) [reply integration/integration.go:121]: I was just curious as to whether the v1 implementation is used anymore. If not could we just convert it to v2? Just think the naming is better in the v1 implementation.
This is a nitpick though...

**Review by rmoles — APPROVED** [review]
Looks good to me! Just 1 nitpick, but happy to merge

File: integration/integration.go (hunk 2 above)
Comment (dmurray-lacework) [reply integration/integration.go:121]: It is no longer used. We will eventually do a clean up in the go-sdk

**Pull request merged**